### ISSUE TITLE
Drawformatted text flips sentence order vertically

### ISSUE THREAD

**Issue by khoi89** [Psychtoolbox-3/Psychtoolbox-3#432]
I am using PsychtoolBox version: 3.0.12 - Flavor: beta - Corresponds to SVN Revision 8020 on MATLAB 2014b build. When using DrawFormattedText, the sentence order is flipped vertically, causing the reading order to go from bottom to top. I did not have this issue a couple weeks ago. I don't know what our IT group did, but this started happening this week. Now - I have debugged this a bit and found that GStreamer is causing this issue. I have uninstalled GStreamer and this vertical ordering issue goes away. I have tried three different builds of GStreamer (tracing back to previous suggestions for versions to use) and this issue persista for GStreamer versions: 1.4.5, 1.8.0, and 1.10.2. Now, I don't know if GStreamer was previously installed on this computer, but since I started debugging it, I noticed that version 1.10.2 was already installed. Any thoughts from the group on what this issue could be?

**Comment by dcnieho** [comment]
Have you updated ptb just before this started happening?

**Comment by khoi89** [comment]
I am not sure since this is not a computer I manage. However, checking the system information, DrawFormattedText.m was last modified on this computer July 17, 2017 and the code in this version was last updated on February 15, 2017. Knowing this, I would probably guess that PTB may have been updated. I guess this would also correspond to why my code worked a couple of weeks ago and did not this week....

Is there a way to check conclusively if PTB was modified on this computer?

**Comment by dcnieho** [comment]
That was just a useful hint. We have done some work on these functions
lately. Could you run a demo coming with ptb called something like
highqualitytextdemo or so (I'm not near a computer until sunday and can't
check the exact name). Do you see the problem there too?

Please leave the recommended version of gstreamer installed. What happens
if you uninstall it is that the high quality text renderer cannot be loaded
and ptb falls back on other old code. You can use
Screen('Preference','TextRenderer',0) to request the old renderer as well,
this may be a workaround for now.

On Jul 27, 2017 5:50 PM, "Khoi Vo" <notifications@github.com> wrote:

I am not sure since this is not a computer I manage. However, checking the
system information, DrawFormattedText.m was last modified on this computer
July 17, 2017 and the code in this version was last updated on February 15,
2017. Knowing this, I would probably guess that PTB may have been updated.
I guess this would also correspond to why my code worked a couple of weeks
ago and did not this week....

Is there a way to check conclusively if PTB was modified on this computer?

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/Psychtoolbox-3/Psychtoolbox-3/issues/432#issuecomment-318403745>,
or mute the thread

**Comment by khoi89** [comment]
The current recommendation for GStreamer is 1.10.2 and that was installed.

The function that you had recommended to test is called DrawHighQualityUnicodeTextDemo.m (that is the closest one I could find based on your recommendation). When I run this with and without gstreamer installed, I get an error after hitting any key to continue as prompted by the function. The error occurs at line 226 of that function:

> Error using Screen
Usage:
>Screen('FrameRect', windowPtr [,color] [,rect] [,penWidth]);
>Error in DrawHighQualityUnicodeTextDemo (line 226)
            Screen('FrameRect', w, [255 255 0], realRect);
 
I have tested another demo called DrawFormattedTextDemo. Without GStreamer, the paragraphs of text appear OK on screen - namely the reading order is still top to bottom. However, when GStreamer is installed, the same issue occurs - the reading order now is flipped vertically with the first line at the bottom and the last line at the top. 

Now - trying out your suggestion for 

> Screen('Preference','TextRenderer',0)

This was able to reverse the issue! With GStreamer installed, I added this line to the DrawFormattedTextDemo.m (after line 25) and the order was correct again. The text that appears on screen is now is similar to when GStreamer was not installed, such as the size of the font on screen and the edginess of the fonts.

I will try to recreate a similar environment on a personal computer to see if I get this same issue. Will also try with an older version of PTB to see, as well. Will report back after trying that this weekend.
